src/main/java/com/google/devtools/build/lib/analysis/actions/SymlinkAction.java

```diff
@@ -214,13 +214,34 @@ public ActionResult execute(ActionExecutionContext actionExecutionContext)
     Path outputPath = getOutputPath(actionExecutionContext);
 
     try {
-      // Delete the empty output directory created prior to the action execution when the output is
-      // a tree artifact. All other actions that produce tree artifacts expect it to exist prior to
-      // their execution. It's not worth complicating ActionOutputDirectoryHelper just to avoid this
-      // small amount of overhead.
-      outputPath.delete();
-
-      outputPath.createSymbolicLink(targetPath);
+      if (getPrimaryOutput().isDirectory()) {
+        // Delete the empty output directory created prior to the action execution when the output
+        // is a tree artifact. All other actions that produce tree artifacts expect it to exist
+        // prior to their execution. It's not worth complicating ActionOutputDirectoryHelper just to
+        // avoid this small amount of overhead.
+        outputPath.delete();
+      }
+      // If the filesystem doesn't support symlinks, it may attempt to copy the file instead or
+      // create a junction (Windows) for a file, which doesn't work if the file hasn't been
+      // downloaded yet. Instead of forcing a download that would block this action, forward the
```

Comment:
Remove the "which doesn't work if the file hasn't been downloaded yet"? My understanding is that a dangling junction is fine - what causes the error is later finding a file (instead of a directory) at the other end of the junction.